docs/reverse.rst

```diff
@@ -2,258 +2,136 @@ Using an external reverse proxy
-Nginx. The ``front`` container will automatically reload Nginx whenever these certificates change.
+  reverse-proxy:
+    # The official v2 Traefik docker image
+    image: traefik:v2.10
+    # Enables the web UI and tells Traefik to listen to docker
+    command:
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--providers.docker.allowEmptyServices=true"
+      - "--entrypoints.web.address=:http"
+      - "--entrypoints.websecure.address=:https"
+      - "--entrypoints.smtp.address=:smtp"
+      - "--entrypoints.submission.address=:submission"
+      - "--entrypoints.submissions.address=:submissions"
+      - "--entrypoints.imap.address=:imap"
+      - "--entrypoints.imaps.address=:imaps"
+      - "--entrypoints.pop3.address=:pop3"
+      - "--entrypoints.pop3s.address=:pop3s"
+      - "--entrypoints.sieve.address=:sieve"
+        #  - "--api.insecure=true"
+      - "--log.level=DEBUG"
+    ports:
+      # The HTTP port
+      - "25:25"
+      - "80:80"
+      - "443:443"
+      - "465:465"
+      - "587:587"
+      - "993:993"
+      - "995:995"
+      - "110:110"
+      - "143:143"
+      - "4190:4190"
+      # The Web UI (enabled by --api.insecure=true)
+      #- "8080:8080"
+    volumes:
+      # So that Traefik can listen to the Docker events
+      - /var/run/docker.sock:/var/run/docker.sock
 
-To set this up, first set ``TLS_FLAVOR=mail`` in your ``.env``. This tells ``mailu/front`` not to try to request certificates using ``letsencrypt``,
-but to read provided certificates, and use them only for mail-protocols, not for ``HTTP``.
-Next, in your ``docker-compose.yml``, comment out the ``port`` lines of the ``front`` section for port ``…:80`` and ``…:443``.
-Add the respective Traefik labels for your domain/configuration, like
+and then add the following to the front section:
 
 .. code-block:: yaml
 
-    labels:
+  labels:
       - "traefik.enable=true"
-      - "traefik.port=80"
-      - "traefik.frontend.rule=Host:$TRAEFIK_DOMAIN"
-
-.. note:: Please don’t forget to add ``TRAEFIK_DOMAIN=[...]`` TO YOUR ``.env``
-
-If your Traefik is configured to automatically request certificates from *letsencrypt*, then you’ll have a certificate
-for ``mail.your.example.com`` now. However, ``mail.your.example.com`` might only be the location where you want the Mailu web-interfaces
-to live — your mail should be sent/received from ``your.example.com``, and this is the ``DOMAIN`` in your ``.env``?
-To support that use-case, Traefik can request ``SANs`` for your domain. The configuration for this will depend on your Traefik version.
 
-Mailu must also be configured with the information what header is used by the reverse proxy for passing the remote
-client IP.  This is configured in mailu.env:
+      # the second part is important to ensure Mailu can get certificates from letsencrypt
+      - "traefik.http.routers.web.rule=Host(`mail.example.com`) || Path(`/.well-known/acme-challenge/`)"
```

Comment:
I know I'm a little late to the party, but shouldn't this be `PathPrefix` instead of `Path` so that it actually matches a challenge? And maybe `||` should be `&&` here as well?

Reply:
Ah yes, PathPrefix it is; I will send a PR tomorrow unless you beat me to it.

The OR is to ensure we catch all hostnames ( client auto configuration, mta-sts, ...)

Reply:
It's all yours. I haven't successfully gotten Traefik working with this method yet, so I wouldn't have a way to test any changes!

Reply:
You need to run master for it to work

Reply:
Ah okay, that explains a lot. I thought I saw some of this PR's changes make its way into 2.0.36 containers, but maybe it's not everything.